components/FileFilters.vue

```diff
@@ -0,0 +1,185 @@
+<script setup lang="ts">
+import { FileObject } from "@supabase/storage-js";
+import type { ModelValue } from "@vuepic/vue-datepicker";
+
+const props = defineProps<{
+  fileList: FileObject[];
+}>();
+
+const isFilterOpen = ref(false);
+const nameFilter = ref("");
+const selectedTypes = ref<string[]>([]);
+const sliderMin = ref(0);
+const sliderMax = ref(5);
+const dates = ref<ModelValue>(null);
+
+const sortTypes = useSortType(undefined, props.fileList);
+
+const isDateValid = computed(() => {
+  return !(
+    Array.isArray(dates.value) && dates.value?.some((el) => el === null)
+  );
+});
```

Comment:
here is checking if the dates ref is Array cuz type ModalValue from vue-datepicker can be array of different data types or just null